lib/index.js

```diff
@@ -1,8 +1,35 @@
 import fetch from 'node-fetch';
 import { promises as fs } from 'fs';
 
-fetch('https://www.reddit.com/r/pics/search.json?q=programming&sort=new')
+export default searchTopics = (topic, sort = "new") => {
+  if (!topic) return; 
+
+  fetch(`https://www.reddit.com/r/pics/search.json?q=${topic}&sort=${sort}`)
   .then(res => res.json())
-  .then((res) => fs.writeFile('data.json', JSON.stringify(res)))
-  .then(() => console.log('Saved Successfully'))
-  .catch(err => console.error(err));
\ No newline at end of file
+  .then(res => res.data.children)
+  .then(posts => posts.map(specifyFields))
+  .then(posts => fs.writeFile('data.json', JSON.stringify(posts, null, '\t')))
+  .catch(err => console.error(err));
+}
+
+const specifyFields = ({ data: 
```

Comment:
How do we handle when the shape of data is not what we expect?
```
{
  data: {
     children: []
  }
}
```

Approach:
1. Enumerate each assumption we are making about our data
* Response is an object
* Response contains a field data
* The field data is an object
* The field data contains a field 'children'
* The field children is an array